model_compression_toolkit/keras/quantizer/mixed_precision/selective_activation_quantizer.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2022 Sony Semiconductors Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import numpy as np
+import tensorflow as tf
+from tensorflow.python.framework.tensor_shape import TensorShape
+from tensorflow_model_optimization.python.core.quantization.keras.quantize_wrapper import QuantizeWrapper
+from tensorflow_model_optimization.python.core.quantization.keras.quantizers import Quantizer
+from typing import Dict, Any, List, Callable
+
+from model_compression_toolkit.common.quantization.candidate_node_quantization_config import \
+    CandidateNodeQuantizationConfig
+from model_compression_toolkit.common.quantization.node_quantization_config import NodeActivationQuantizationConfig
+
+
+class SelectiveActivationQuantizer(Quantizer):
+    """
+    Quantizer that can use different quantized weights on-the-fly.
+    The general idea behind this kind of quantizer is that it gets the float tensor to quantize
+    when initialize, it quantizes the float tensor in different bitwidths, and every time it need to return a
+    quantized version of the float weight, it returns only one quantized weight according to an "active"
+    index - the index of a candidate weight quantization configuration from a list of candidates that was passed
+    to the SelectiveWeightsQuantizer when it was initialized.
+    The "active" index can be configured as part of the SelectiveWeightsQuantizer's API, so a different quantized
+    weight can be returned in another time.
+    """
+
+    def __init__(self,
+                 node_q_cfg: List[CandidateNodeQuantizationConfig]):
+        """
+        Init a selective quantizer.
+
+        Args:
+            node_q_cfg: Quantization configuration candidate of the node that generated the layer that will
+                use this quantizer.
+        """
+        self.node_q_cfg = node_q_cfg
+
+        # Use the node's quantizer. The SelectiveWeightsQuantizer is supported only if
+        # all node's qc candidates use the same quantizer.
+        quantizer_fn = self.node_q_cfg[0].activation_quantization_cfg.activation_quantization_fn
+        for qc in self.node_q_cfg:
+            assert qc.activation_quantization_cfg.activation_quantization_fn == quantizer_fn
```

Comment:
Why do we constrain it? Unless it's temporary until upcoming PRs

Reply:
No, it's a mistake, I'll remove it